tests/testthat/test-module_teal.R

```diff
@@ -1,97 +1,916 @@
+      moduleServer(id, function(input, output, session) {
+        reactive({
+          stop("oh no!")
+        })
+      })
+    }
+  ),
+  iris = teal_data_module(
+    ui = function(id) NULL,
+    server = function(id, data) {
+      moduleServer(id, function(input, output, session) {
+        reactive({
+          within(data(), iris <- head(iris))
+        })
+      })
+    }
+  ),
+  mtcars = teal_data_module(
+    ui = function(id) NULL,
+    server = function(id, data) {
+      moduleServer(id, function(input, output, session) {
+        reactive({
+          within(data(), mtcars <- head(mtcars))
+        })
+      })
+    }
+  )
+)
+
+testthat::describe("srv_teal arguments", {
+  testthat::it("accepts data to be teal_data", {
+    testthat::expect_no_error(
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = teal.data::teal_data(iris = iris),
+          modules = modules(example_module())
+        ),
+        expr = NULL
+      )
+    )
+  })
+
+  testthat::it("accepts data to be teal_data_module returning reactive teal_data", {
+    testthat::expect_no_error(
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = teal_data_module(ui = function(id) NULL, server = function(id) reactive(teal_data(iris = iris))),
+          modules = modules(example_module())
+        ),
+        expr = NULL
+      )
+    )
+  })
+
+  testthat::it("accepts data to a reactive or reactiveVal teal_data", {
+    testthat::expect_no_error(
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = reactive(teal_data(iris = iris)),
+          modules = modules(example_module())
+        ),
+        expr = NULL
+      )
+    )
+
+    reactive_val <- reactiveVal(teal_data(iris = iris))
+    testthat::expect_no_error(
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = reactive_val,
+          modules = modules(example_module())
+        ),
+        expr = NULL
+      )
+    )
+  })
+
+  testthat::it("fails when data is not teal_data or teal_data_module", {
+    testthat::expect_error(
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = data.frame(),
+          modules = modules(example_module())
+        ),
+        expr = NULL
+      ),
+      "Must inherit from class 'teal_data'/'teal_data_module'/'reactive'/'reactiveVal'"
+    )
+  })
+
+  testthat::it("app fails when teal_data_module doesn't return a reactive", {
+    testthat::expect_error(
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = teal_data_module(ui = function(id) NULL, server = function(id) teal_data(iris = iris)),
+          modules = modules(example_module())
+        ),
+        expr = {
+          session$flushReact()
+        }
```

Comment:
```suggestion
        expr = NULL
```
Is the flush needed? I think the error is thrown even before the first reactive cycle is complete.